sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SymbolicExecution/ExplodedGraphTests.cs

```diff
@@ -97,50 +84,28 @@ public void ExplodedGraph_SequentialInput()
                     }
                 };
 
-            explodedGraph.Walk();
-
-            explorationEnded.Should().BeTrue();
-            numberOfProcessedInstructions.Should().Be(9);
-            numberOfExitBlockReached.Should().Be(1);
+            context.Walk(9);
```

Comment:
I think it would be better to separate the `Walk`step from the assertion step, that way it is clear what we do in each method:
```
context.Walk();

context.HasWalked(9);
```
I would find the above easier to read, and it would not need the comments `Special case with manual checks` as it would be clearly separated

Reply:
There are 3 more checks in the `Walk` itself. And I wanted to enforce all these checks for every test (except special cases) to maximize coverage. Newly added tests tended to miss some of those checks.

I'd prefer just to rename the method for something more explicit. Like `WalkWithInstructions(9)` as there already exists `WalkWithExitBlocks`